### PR TITLE
[SYCL] Add E2E tests for device code instrumentation

### PR DESCRIPTION
**Pull request by AGindinson**
Add a couple tests with execute-only kernels to ensure that ITT
annotations' generation doesn't break the compilation/execution.

This aims to test intel/llvm#4615.

Signed-off-by: Artem Gindinson <artem.gindinson@intel.com>

### CODE REVIEW

**Comment by AGindinson** [comment]
The tests are non-breaking even w/o the product changes, so committing them and re-running Precommit for intel/llvm#4615 is a possibility. EDIT: Actually, the test changes **are** breaking, as the product changes include the fix for the required CLI option...

**Review by vladimirlaz — DISMISSED** [review]

**Review by MrSidims — DISMISSED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -23,6 +23,7 @@ SYCL/DeviceCodeSplit @AlexeySachkov  @Fznamznon
 
 # Device library
 SYCL/DeviceLib @vzakhari
+SYCL/DeviceLib/ITTAnnotations @vzahkari @MrSidims @AGindinson
```
Comment (vzakhari) [inline .github/CODEOWNERS]: typo: vzahkari -> vzakhari

**Comment by AGindinson** [comment]
@vzakhari @vladimirlaz The CI has finally picked the correct nightly build of the compiler, so the patch should be good for the final review.

**Review by vladimirlaz — DISMISSED** [review]

File: SYCL/DeviceLib/ITTAnnotations/barrier.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// UNSUPPORTED: cuda
```
Comment (vladimirlaz) [inline SYCL/DeviceLib/ITTAnnotations/barrier.cpp]: why ITT is unsupported on CUDA? Is is due to AOT nature of the build?
is it supported on HIP?

Comment (AGindinson) [reply SYCL/DeviceLib/ITTAnnotations/barrier.cpp]: It has to do with lack of device library support. AMDGCN should actually be excluded as well - I believe this hasn't been caught by CI due to the current lack of comprehensive HIP testing.

**Comment by vladimirlaz** [comment]
the fix for the failure is under intel/llvm-test-suite#494

**Review by vladimirlaz — APPROVED** [review]

**Review by vzakhari — APPROVED** [review]

**Pull request merged**